### PR TITLE
feat: enforce use of variables for colors (hex, rgb, hsl and named)

### LINKED ISSUE #56: Error out on shorthands (stylelint-declaration-strict-value)
We are using `stylelint-declaration-strict-value` to forbid the usage of plain values in certain props, like `color`.

However, it's not working correctly for `background` and other shorthands. Perhaps we can use [`declaration-property-value-blacklist`](https://stylelint.io/user-guide/rules/declaration-property-value-blacklist) to fix that.

### LINKED ISSUE #60: Lax rules that force the use of variables and units to warnings

### PR DESCRIPTION
**Pull request by jdiogopeixoto**
Changes on this PR:

- Enforce use of variables for all named colors - `color-named` rule
- Blacklist hex, rgb(a) and hsl(a) colors for some properties - `declaration-property-value-blacklist` rule
- Remove color values check from `declaration-strict-value`
- Change from `error` to `warning` all the rules that enforce the use of variables (#60)

Closes #56 
Closes #60

### CODE REVIEW

File: rules/base.js [hunk 1]
```diff
@@ -17,6 +17,9 @@ module.exports = {
         'block-opening-brace-space-before': 'always',
         'color-hex-case': 'lower',
         'color-hex-length': 'short',
+        'color-named': ['never', {
+            'ignoreProperties': ['/^(--).*/'],
```
Comment (satazor) [inline rules/base.js]:
```suggestion
            'ignoreProperties': ['/^--\w/'],
```

File: rules/base.js [hunk 2]
```diff
@@ -38,6 +41,9 @@ module.exports = {
         'declaration-colon-newline-after': 'always-multi-line',
         'declaration-colon-space-after': 'always-single-line',
         'declaration-colon-space-before': 'never',
+        'declaration-property-value-blacklist': {
+            '/^(?!--).*(background|color|fill|stroke|border|shadow|outline).*/': ['/#([a-f0-9]{3,})*/i', '/(rgb|hsl)a?\\(/'],
```
Comment (satazor) [inline rules/base.js]:
```suggestion
            '/^(?!--).*(background|color|fill|stroke|border|shadow|outline)/': ['/#([a-f0-9]{3,})*/i', '/(rgb|hsl)a?\\(/'],
```

Comment (satazor) [reply rules/base.js]: Also, do we need the `.*` after `(?!--)`?

**Comment by satazor** [comment]
We have some conflicts that need to be taken care of because the other PR got merged.

File: rules/base.js (hunk 2 above)
Comment (jdiogopeixoto) [reply rules/base.js]: Yes, for cases like `text-decoration-color`

Comment (satazor) [reply rules/base.js]: I removed my suggestion then. I still think we may replace the trailing `.*` with `\b` instead.

Comment (jdiogopeixoto) [reply rules/base.js]: Replacing `.*` with `\b` only works when the property starts with one of the names on the list - eg. `color`, `background`.
For cases likes `*-color` it doesn't work.

Comment (satazor) [reply rules/base.js]: I was referring to the trailing `.*`, not the initial one.

**Comment by satazor** [comment]
Looks good. @nlfonseca do you want to take a look at this too?

**Pull request merged**